### PR TITLE
Add function attachInterrupt. 

### PR DESCRIPTION
**Pull request by ihere1**
Usage: attachInterrupt(PA_4, blink, FALLING);

### CODE REVIEW

**Comment by ihere1** [comment]
https://github.com/Microsoft/devkit-sdk/issues/789

File: AZ3166/src/cores/arduino/AttachInterrupt.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "AttachInterrupt.h"
+#include "wiring_constants.h"
+
+static PinName validPinames[16] = {PB_0, NC, PB_2, PB_3, PA_4, PA_5, PB_6, PB_7, PB_8, PB_9, PA_10, NC, NC, PB_13, PB_14, PB_15};
```
Comment (ArthurMa1978) [inline AZ3166/src/cores/arduino/AttachInterrupt.cpp]: Comment all the valid Pin name in the header file and provide more descriptions

File: AZ3166/src/cores/arduino/AttachInterrupt.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+**
+* @param    pin               The PinName of digital pin.
+* @param    ISR               The callback function.
+* @param    mode              Triggered event type, support CHANGE, RISING, FALLING now.
+*
+* @return   Return 0 on success. Return -1 on fail.
+*           The failure might be caused by unsupported PinName.
+*/
+int attachInterrupt(PinName pin, Callback<void()> ISR, int mode);
+
```
Comment (ArthurMa1978) [inline AZ3166/src/cores/arduino/AttachInterrupt.h]: please work with @v-fional to add unit test for this new feature

File: AZ3166/src/cores/arduino/AttachInterrupt.cpp (hunk 1 above)
Comment (ihere1) [reply AZ3166/src/cores/arduino/AttachInterrupt.cpp]: Done.

**Pull request merged**